File: leetcode/easy/1228-missing-number-in-arithmetic-progression/1228-missing-number-in-arithmetic-progression.kt
class Solution {
    fun missingNumber(arr: IntArray): Int {
        val diff = (arr[arr.size - 1] - arr[0]) / arr.size

        for (i in 1 until arr.size) {
            if (arr[i] - arr[i - 1] != diff) {
                return arr[i - 1] + diff
            }
        }

        return arr[0]
    }
}

File: leetcode/easy/1228-missing-number-in-arithmetic-progression/1228-missing-number-in-arithmetic-progression.py
class Solution:
    def missingNumber(self, arr: List[int]) -> int:
        diff = (arr[-1] - arr[0]) // len(arr)

        for i in range(1, len(arr)):
            if arr[i] != arr[i-1] + diff:
                return arr[i-1] + diff

        return arr[0]
